src/main/java/de/mediathekview/mserver/crawler/zdf/tasks/ZdfTaskBase.java

```diff
@@ -0,0 +1,70 @@
+package de.mediathekview.mserver.crawler.zdf.tasks;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import de.mediathekview.mserver.crawler.basic.AbstractCrawler;
+import de.mediathekview.mserver.crawler.basic.AbstractRestTask;
+import de.mediathekview.mserver.crawler.basic.CrawlerUrlDTO;
+import de.mediathekview.mserver.crawler.zdf.ZdfConstants;
+import java.lang.reflect.Type;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public abstract class ZdfTaskBase<T, D extends CrawlerUrlDTO> extends AbstractRestTask<T, D> {
+
+  private static final Logger LOG = LogManager.getLogger(ZdfTaskBase.class);
+
+  private final GsonBuilder gsonBuilder;
+
+  public ZdfTaskBase(AbstractCrawler aCrawler,
+      ConcurrentLinkedQueue<D> aUrlToCrawlDtos, Optional<String> aAuthKey) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "aUrlToCrawlDtos" object should be an interface such as "Queue" rather than the implementation "ConcurrentLinkedQueue". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)